modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,460 @@
+ */
+
+/**
+ * @typedef {Object} AnalyticsCache
+ * @property {string} pid Partner ID
+ * @property {Object<string, Auction>} auctions
+ * @property {Object<string, Bid[]>} bidsWon
+ */
+
+/**
+ * @typedef {Object} Auction - Parsed auction data
+ * @property {AdUnit[]} adUnits
+ * @property {string} auctionId
+ * @property {string[]} userIds
+ */
+
+/**
+ * @typedef {`${number}x${number}`} AdUnitSize
+ */
+
+/**
+ * @typedef {('banner'|'native'|'video')} AdUnitMediaType
+ */
+
+/**
+ * @typedef {Object} BidResponse
+ * @property {number} cpm
+ * @property {string} cur
+ * @property {number} [cpmOrig]
+ * @property {number} cpmFloor
+ * @property {AdUnitMediaType} mediaType
+ * @property {AdUnitSize} size
+ */
+
+/**
+ * @typedef {Object} Bid - Parsed bid data
+ * @property {string} bidder
+ * @property {string} bidId
+ * @property {string} source
+ * @property {string} status
+ * @property {BidResponse} [bidResponse]
+ * @property {1|0} [hasWon]
+ */
+
+/**
+ * @typedef {Object} AdUnit - Parsed adUnit data
+ * @property {string} transactionId - Primary key for *this* auction/adUnit combination
+ * @property {string} adUnitCode
+ * @property {string} slotId - Equivalent to GPID. (Note that
+ * GPID supports adUnits where multiple units have the same `code` values
+ * by appending a `#UNIQUIFIER`. The value of the UNIQUIFIER is likely to be the div-id,
+ * but, if div-id is randomized / unavailable, may be something else like the media size)
+ * @property {Array<AdUnitMediaType>} mediaTypes
+ * @property {Array<AdUnitSize>} sizes
+ * @property {Array<Bid>} bids
+ */
+
+/**
+ * After the first transaction begins, wait until all transactions are complete
+ * before calling `onComplete`. If the timeout is reached before all transactions
+ * are complete, send the report anyway.
+ *
+ * Use this to track all transactions per auction, and send the report as soon
+ * as all adUnits have been won (or after timeout) even if other bid/auction
+ * activity is still happening.
+ */
+class TransactionManager {
+  /**
+   * Milliseconds between activity to allow until this collection automatically completes.
+   * @type {number}
+   */
+  #sendTimeout;
+  #timeoutId;
+  #transactions = new Set();
+  #completed = new Set();
+  #onComplete;
+
+  constructor({ timeout, onComplete }) {
+    this.#sendTimeout = timeout;
+    this.#onComplete = onComplete;
+  }
+
+  initiate(transactionId) {
+    this.#transactions.add(transactionId);
+    this.#restartSendTimeout();
+  }
+
+  complete(transactionId) {
+    if (!this.#transactions.has(transactionId)) {
+      log.warn(`transactionId "${transactionId}" was not found. No transaction to mark as complete.`);
+      return;
+    }
+
+    this.#transactions.delete(transactionId);
+    this.#completed.add(transactionId);
+    log.info(`Marked transaction "${transactionId}" as complete. ${this.#transactions.size} transaction(s) pending.`, {pending: this.#transactions, completed: this.#completed});
```

Comment:
This puts a hefty amount of text in the console that's only really interesting or meaningful to our dev team. If Prebid offered a `logDebug` or `logTrace` level then this would be a good fit for it, but it looks like they don't. We can leave this in for now to facilitate internal testing, but check the other Prebid modules to see whether they're using `log.info` in this way, and if not then consider whittling down these `log.info` statements before the final push to Prebid.